dev/source/docs/learning-ardupilot-threading.rst

```diff
@@ -225,188 +225,188 @@ card and connecting to the USB console:
 sketch.
-
+
 If you look inside that file you will see a small table with a set of 3
 scheduling tasks. Associated with each task are two numbers. The table
 looks like this:
-
+
 ::
-
+
     static const AP_Scheduler::Task scheduler_tasks[] PROGMEM = {
      { ins_update, 1, 1000 },
      { one_hz_print, 50, 1000 },
      { five_second_call, 250, 1800 },
     };
-
+
 The first number after each function name is the call frequency, in
 units controlled by the ins.init() call. For this example sketch the
 ins.init() uses RATE_50HZ, so each scheduling step is 20ms. That means
 the ins_update() call is made every 20ms, the one_hz_print() function
 is called every 50 times (ie. once a second) and the
 five_second_call() is called every 250 times (ie. once every 5
 seconds).
-
+
 The third number is the maximum time that the function is expected to
 take. This is used to avoid making the call unless there is enough time
 left in this scheduling run to run the function. When scheduler.run() is
 called it is passed the amount of time (in microseconds) available for
 running tasks, and if the worst case time for this task would mean it
 wouldn't fit before that time runs out then it won't be called.
-
+
 Another point to look at closely is the ins.wait_for_sample() call.
 That is the "metronome" that drives the scheduling in ArduPilot. It
 blocks execution of the main vehicle thread until a new IMU sample is
 available. The time between IMU samples is controlled by the arguments
 to the ins.init() call.
-
+
 Note that tasks in AP_Scheduler tables must have the following
 attributes:
-
+
 -  they should never block (except for the ins.update() call)
 -  they should never call sleep functions when flying (an autopilot,
    like a real pilot, should never sleep while flying)
 -  they should have predictable worst case timing
-
+
 You should now go and modify the Scheduler_test example and add in your
 own tasks to run. Try adding tasks that do the following:
-
+
 -  read the barometer
 -  read the compass
 -  read the GPS
 -  update the AHRS and print the roll/pitch
-
+
 Look at the example sketches for each library that you worked with
 earlier in this tutorial to understand how to use each sensor library.
-
+
 Semaphores
 ==========
-
+
 When you have multiple threads (or timer callbacks) you need to ensure
 that data structures shared by the two logical threads of execution are
 updated in a way that prevents corruption. There are 3 principle ways of
 doing this in ArduPilot - semaphores, lockless data structures and the
 PX4 ORB.
-
+
 AP_HAL Semaphores are just wrappers around whatever semaphore system is
 available on the specific platform, and provide a simple mechanism for
 mutual exclusion. For example, I2C drivers can ask for the I2C bus
 semaphore to ensure that only one I2C device is used at a time.
-
-Go and have a look at the hmc5883 driver in

```

Comment:
@tridge @rmackay9 Can you confirm this change is OK and merge if it is (FYI this comment is a line comment for the only relevant change in this merge).